integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/MessageInController.java

```diff
@@ -0,0 +1,193 @@
+
+@RestController
+public class MessageInController {
+
+    private static final Logger log = LoggerFactory.getLogger(MessageInController.class);
+
+    @Autowired
+    private IncomingConversationResourceRepository repo;
+
+    @Autowired
+    private IntegrasjonspunktProperties props;
+
+    @RequestMapping(value = "/in/messages", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity getIncomingMessages(
+            @RequestParam(value = "messagetypeId", required = false) String messagetypeId,
+            @RequestParam(value = "conversationId", required = false) String conversationId,
+            @RequestParam(value = "senderId", required = false) String senderId) {
+
+        if (!isNullOrEmpty(conversationId)) {
+            Optional<IncomingConversationResource> resource = Optional.ofNullable(repo.findOne(conversationId));
+            if (resource.isPresent()) {
+                return ResponseEntity.ok(resource.get());
+            }
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found.");
+        }
+
+        List<IncomingConversationResource> resources;
+        if (!isNullOrEmpty(messagetypeId)) {
+            if (!isNullOrEmpty(senderId)) {
+                resources = repo.findByMessagetypeIdAndSenderId(messagetypeId, senderId);
+            } else {
+                resources = repo.findByMessagetypeId(messagetypeId);
+            }
+        } else {
+            if (!isNullOrEmpty(senderId)) {
+                resources = repo.findBySenderId(senderId);
+            } else {
+                resources = Lists.newArrayList(repo.findAll());
+            }
+        }
+        return ResponseEntity.ok(resources);
+    }
+
+    @RequestMapping(value = "/in/messages/pop", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity popIncomingMessages(
+            @RequestParam(value = "messagetypeId", required = false) String messagetypeId) throws FileNotFoundException {
+
+        Optional<IncomingConversationResource> resource;
+        if (isNullOrEmpty(messagetypeId)) {
+            resource = repo.findFirstByOrderByLastUpdateAsc();
+        } else {
+            resource = repo.findFirstByMessagetypeIdOrderByLastUpdateAsc(messagetypeId);
+        }
+
+        if (resource.isPresent()) {
+            String fileName = resource.get().getFileRefs().get(0);
+            String filedir = props.getNextbest().getFiledir();
+            if (!filedir.endsWith("/")) {
+                filedir = filedir+"/";
+            }
+            filedir = filedir+resource.get().getConversationId()+"/";
+            File file = new File(filedir+fileName);
+
+            InputStreamResource isr = new InputStreamResource(new FileInputStream(file));
+
+            repo.delete(resource.get());
+
+            return ResponseEntity.ok()
+                    .header("Content-Disposition", "attachement; filename="+fileName)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "attachement; filename=" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Content-Disposition" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)